openedx/core/djangoapps/schedules/resolvers.py

```diff
@@ -455,57 +454,60 @@ def send(self):
                 self.async_send_task.apply_async((self.site.id, str(msg)), retry=False)
 
     def get_schedules(self):
-        course_key = CourseKey.from_string(self.course_id)
+        """
+        Grabs possible schedules that could receive a Course Next Section Update and if a
+        next section highlight is applicable for the user, yields information needed to
+        send the next section highlight email.
+        """
         target_date = self.target_datetime.date()
-        schedules = get_schedules_with_due_date(course_key, target_date).filter(
+        course_duration = get_expected_duration(self.course_id)
+        schedules = Schedule.objects.select_related('enrollment').filter(
             self.experience_filter,
             active=True,
+            enrollment__course_id=self.course_id,
             enrollment__user__is_active=True,
+            start_date__gte=target_date - course_duration,
+            start_date__lt=target_date,
         )
 
         template_context = get_base_template_context(self.site)
         for schedule in schedules:
-            enrollment = schedule.enrollment
             course = schedule.enrollment.course
-            user = enrollment.user
+            user = schedule.enrollment.user
             start_date = max(filter(None, (schedule.start_date, course.start)))
             LOG.info('Received a schedule for user {} in course {} for date {}'.format(
-                user.username,
-                self.course_id,
-                target_date,
+                user.username, self.course_id, target_date,
             ))
 
             try:
                 week_highlights, week_num = get_next_section_highlights(user, course.id, start_date, target_date)
+                # (None, None) is returned when there is no section with a due date of the target_date
+                if week_highlights is None:
+                    continue
             except CourseUpdateDoesNotExist as e:
                 LOG.warning(e.args)
-                LOG.warning(
-                    'Weekly highlights for user {} of course {} does not exist or is disabled'.format(
-                        user, course.id
-                    )
-                )
```

Comment:
Now that we log out the more descriptive message, I felt like I could remove this. Maybe I should use a prefix though so they are easy to find. Let me update.